### PR TITLE
add ConsignmentEndpoint enum

### LINKED ISSUE #2: modify `consignment_endpoints` parameter
Looking at the `Invoice` struct I see a [`consignment_endpoints` parameter](https://github.com/LNP-BP/invoices/blob/master/rust/src/base.rs#L126) that is defined as `Vec<NodeAddr>`. `NodeAddr` is a struct with an `id` (`NodeId`) and an `addr` (`InetSocketAddr`) and is documented as `Internet P2P node address`.

We believe that `consignment_endpoints` should be changed to something as (wording can be discussed) `Vec<ConsignmentEndpoint>` where `ConsignmentEndpoint` is defined as an enum that can be `Storm(NodeAddr)` or `RestProxy(Url)`. In this way in the future it will be easy and backwards-compatible to add other "protocols" that allow the transfer of consignments.

[rgb-proxy-server](https://github.com/grunch/rgb-proxy-server) is an example of such "rest proxy", I plan to add documentation on "protocol" APIs so that the proxy can be implemented also in other programming languages.

### PR DESCRIPTION
**Pull request by zoedberg**
This PR introduces a `ConsignmentEndpoint` enum, currently defining 2 possible exchange medium protocols: `Storm` (https://github.com/Storm-WG/storm-spec) and `RgbHttpJsonRpc` (https://github.com/RGB-Tools/rgb-http-json-rpc).

For the `RgbHttpJsonRpc` protocol URL I've used a `String`, because `url::Url` doesn't seem supported by strict enconding. I see other string parameters representing URLs (e.g. `price_provider`), so I don't think this is an issue, but @dr-orlovsky let me know if you think there are better solutions.

Moreover in the `add_consignment_endpoint` method I've removed the call to `is_rgb`, since that method, in order to decide if the consignment is for RGB, checks whether the invoice specifies a contract ID. But we see use cases where an RGB invoice could not contain asset information but would like to include consignment endpoints and removing that call achieves that.

Closes #2

### CODE REVIEW

File: rust/src/base.rs [hunk 1]
```diff
@@ -484,6 +481,64 @@ impl<'a> Iterator for BeneficiariesIter<'a> {
...
+#[display(inner)]
+#[non_exhaustive]
+pub enum ConsignmentEndpoint {
+    /// Storm protocol
+    #[display("storm:{0}")]
+    Storm(NodeAddr),
+
+    /// RGB HTTP JSON-RPC protocol
+    #[display("rgbhttpjsonrpc:{0}")]
+    RgbHttpJsonRpc(String), // Url,
```
Comment (dr-orlovsky) [inline rust/src/base.rs:511]: Nit: I think just `RgbJsonRpc` would work, `Http` part is obvious for `JsonRpc` protocol.

**Review by dr-orlovsky — APPROVED** [review]
ACK https://github.com/LNP-BP/invoices/commit/a608636da5611e2518c47caeba8cb889915595d9

This looks good for me; I'll merge it as is, but there are several questions I'd like to discuss. I will post them as a separate issues in this repo

**Comment by fedsten** [comment]
@dr-orlovsky since this PR has been merged, could you make a new release on crates of  `lnpbp-invoice` (e.g. 0.8.0-rc.2). This would be helpful to test it on Iris without git dependencies on rgb-lib.

**Comment by zoedberg** [comment]
Sorry, I've just noticed a bug in this PR. I will fix it as soon as possible and then a release will be very appreciated.

**Comment by dr-orlovsky** [comment]
Waiting for the confirmation that #6 solves the problem - and then will release invoice crate

**Comment by zoedberg** [comment]
#6 solves the problem. @dr-orlovsky you can release, thank you

**Comment by dr-orlovsky** [comment]
@zoedberg published

**Pull request merged**